functional-tests/functional/dart/main.dart

```diff
@@ -102,6 +104,17 @@ final _allTests = [
   StructsWithMethodsTests.main
 ];
 
-void main() {
+String _getLibraryPath(String nativeLibraryName) {
+  if (Platform.isWindows) return 'lib${nativeLibraryName}.dll';
+  if (Platform.isMacOS || Platform.isIOS) return 'lib${nativeLibraryName}.dylib';
+  return 'lib${nativeLibraryName}.so';
+}
+
+void main(List<String> arguments) {
+  final libraryName = arguments.isEmpty ? _getLibraryPath('functional') : arguments[0];
+  __lib.LibraryContext.init(__lib.IsolateOrigin.main, nativeLibraryPath: libraryName);
+
   _allTests.forEach((testCase) => testCase());
+  CallbacksMultithreadedTests.main(libraryName);
```

Comment:
I think these two lines need to be swapped. Multithreaded tests call `init()` and `release()` on their own.